sdks/java/io/fileschematransform/src/main/java/org/apache/beam/sdk/io/fileschematransform/FileWriteSchemaTransformConfiguration.java

```diff
@@ -0,0 +1,215 @@
+
+    /** Uses the given {@link org.apache.beam.sdk.io.ShardNameTemplate} for naming output files. */
+    public abstract Builder setShardNameTemplate(String value);
+
+    /** Configures the filename suffix for written files. */
+    public abstract Builder setFilenameSuffix(String value);
+
+    /** Configures extra details related to writing CSV formatted files. */
+    public abstract Builder setCsvConfiguration(CsvConfiguration value);
+
+    /** Configures extra details related to writing Parquet formatted files. */
+    public abstract Builder setParquetConfiguration(ParquetConfiguration value);
+
+    /** Configures extra details related to writing XML formatted files. */
+    public abstract Builder setXmlConfiguration(XmlConfiguration value);
+
+    public abstract FileWriteSchemaTransformConfiguration build();
+  }
+
+  /** Configures extra details related to writing CSV formatted files. */
+  @DefaultSchema(AutoValueSchema.class)
+  @AutoValue
+  public abstract static class CsvConfiguration {
+
+    /** Content written to the top of every sharded file prior to the header. */
+    @Nullable
+    public abstract String getPreamble();
+
+    /**
+     * The format of the written CSV file. See <a
+     * href="https://commons.apache.org/proper/commons-csv/apidocs/org/apache/commons/csv/CSVFormat.html">org.apache.commons.csv.CSVFormat</a>
+     * for allowed values, stringified in lowercase. Defaults to <a
+     * href="https://commons.apache.org/proper/commons-csv/apidocs/org/apache/commons/csv/CSVFormat.html#DEFAULT">CSVFormat.DEFAULT</a>
+     */
+    @Nullable
+    public abstract String getCsvFormat();
+
+    @AutoValue.Builder
+    public abstract static class Builder {
+
+      /** Content written to the top of every sharded file prior to the header. */
+      public abstract Builder setPreamble(String value);
+
+      /**
+       * The format of the written CSV file. See <a
+       * href="https://commons.apache.org/proper/commons-csv/apidocs/org/apache/commons/csv/CSVFormat.html">org.apache.commons.csv.CSVFormat</a>
+       * for allowed values, stringified in lowercase. Defaults to <a
+       * href="https://commons.apache.org/proper/commons-csv/apidocs/org/apache/commons/csv/CSVFormat.html#DEFAULT">CSVFormat.DEFAULT</a>
+       */
+      public abstract Builder setCsvFormat(String value);
+
+      public abstract CsvConfiguration build();
+    }
+  }
+
+  /** Configures extra details related to writing Parquet formatted files. */
+  @DefaultSchema(AutoValueSchema.class)
+  @AutoValue
+  public abstract static class ParquetConfiguration {
+
+    /**
+     * Specifies compression codec. See org.apache.parquet.hadoop.metadata.CompressionCodecName for
+     * allowed names, stringified in lowercase format.
+     */
+    public abstract String getCompressionCodecName();
+
+    /** Specify row-group size; if not set or zero, a default is used by the underlying writer. */
+    public abstract Integer getRowGroupSize();
```

Comment:
does this property have a default value? It should : ) - usually I expect users to not be so concerned with this. wdyt?

Reply:
I copied the Javadoc from [ParquetIO.Sink#withRowGroupSize](https://beam.apache.org/releases/javadoc/current/org/apache/beam/sdk/io/parquet/ParquetIO.Sink.html#withRowGroupSize-int-) and deferring to its documented functionality.

Reply:
makes sense. thanks!